chia/data_layer/data_store.py

```diff
@@ -289,26 +289,42 @@ async def _insert_ancestor_table(
                     "tree_id": tree_id,
                     "generation": generation,
                 }
-                cursor = await writer.execute(
-                    "SELECT * FROM ancestors WHERE hash == :hash AND generation == :generation AND tree_id == :tree_id",
-                    {"hash": hash, "generation": generation, "tree_id": tree_id},
-                )
-                result = await cursor.fetchone()
-                if result is None:
+                try:
                     await writer.execute(
                         """
                         INSERT INTO ancestors(hash, ancestor, tree_id, generation)
                         VALUES (:hash, :ancestor, :tree_id, :generation)
                         """,
                         values,
                     )
-                else:
+                except aiosqlite.IntegrityError as e:
+                    if not e.args[0].startswith("UNIQUE constraint"):
+                        # UNIQUE constraint failed: ancestors.hash, ancestors.tree_id, ancestors.generation
+                        raise
+
+                    async with writer.execute(
+                        """
```

Comment:
do we really need this ? looks like we already checked if we got the expected error, at this point are we validating sqlite is doing its work ?